### PR TITLE
Serveral fixes and WP Code Style

### PR DESCRIPTION
**Pull request by bueltge**
Hi.
Thanks a lot for this design, plugin.

I have problems in the multiste environment if I change the location of the wp-plugins directory to load the stylesheet file. I fxed this and include also lot of changes to simplify the code, write code in the rules of the WP Code Style and now its load only if WP is ready for plugins and not in the php environment. Thats makes more solid.

Small hint, you should use php 5.6 for the minimal php version, older verisons are not supported in the security context. Also a small hint to remove the index.html in the plugin directory. It is only a task on the server side to load them and can't use, because she is empty. It is not helpful in the security context, is more an task on the server and installation.

All the best.

### CODE REVIEW

**Comment by danieltj27** [comment]
Hi @bueltge, thank you for spending time on this. There's quite a lot of changes here and I think the best course of action is to focus on smaller bits and iterate over time.

**Comment by bueltge** [comment]
Thanks for your fast response. I hope the commits are really simple enough and short to understand his goal. However I think a test should helps you in a effektive way. In the context of multisite and changes of the directories wp-plugins, wp-content ([like this way](https://github.com/bueltge/WordPress-Starter/blob/master/wp-config.php#L75)) it works now really great.

**Comment by danieltj27** [comment]
No worries, let's gradually integrate a few of these changes over time. I'll leave some comments on the commits and we can go from there. I'm also going to release a new version with what we have so far.

File: dark-mode.php [hunk 1]
```diff
@@ -7,80 +7,98 @@
...
  * Author URI: https://www.danieltj.co.uk/
  * Text Domain: dark-mode
- * Version: 1.8.3
+ * Version: 1.8.4
  */
 
-// No thank you
-if ( ! defined( 'ABSPATH' ) ) die();
+// No thank you.
+if ( ! defined( 'ABSPATH' ) ) {
```
Comment (danieltj27) [inline dark-mode.php]: We can leave this bit as it is. Whilst it does use shorthand PHP, Core does do this too.

File: dark-mode.php [hunk 2]
```diff
@@ -7,80 +7,98 @@
...
+	die();
+};
 
-new Dark_Mode;
+/**
+ * Run if WP is ready and will load plugins.
+ */
+add_action( 'plugins_loaded', array( 'Dark_Mode', 'init' ) );
```
Comment (danieltj27) [inline dark-mode.php]: The plugin is loaded when `plugins_loaded` is fired so we don't need to take this out of the class.

File: dark-mode.php [hunk 3]
```diff
@@ -7,80 +7,98 @@
...
 	 * @since 1.0
 	 * @since 1.1 Changed admin_enqueue_scripts hook to 99 to override admin colour scheme styles.
 	 * @since 1.3 Added hook for the Feedback link in the toolbar.
 	 * @since 1.8 Added filter for the plugin table links and removed admin toolbar hook.
-	 * 
+	 *
 	 * @return void
 	 */
-	public function __construct() {
+	public function __construct() {}
```
Comment (danieltj27) [inline dark-mode.php]: See above comment about `plugins_loaded`. It's also best to not have empty functions.

File: dark-mode.php [hunk 4]
```diff
@@ -7,80 +7,98 @@
...
+
+	/**
+	 * Function which hooks into WordPress Core.
+	 * Load all resources in the WP environment.
+	 *
+	 * @since 1.8.4
+	 */
+	public function init() {
```
Comment (danieltj27) [inline dark-mode.php]: This can be merged back into the constructor function.

File: dark-mode.php [hunk 5]
```diff
@@ -7,80 +7,98 @@
...
 	 * @since 1.0
-	 * 
+	 *
 	 * @return void
 	 */
 	public static function load_text_domain() {
 
-		load_plugin_textdomain( 'dark-mode', false, untrailingslashit( dirname( __FILE__ ) ) . '/languages' );
-
+		load_plugin_textdomain(
```
Comment (danieltj27) [inline dark-mode.php]: Let's keep the function on one line for readability.

File: dark-mode.php [hunk 6]
```diff
@@ -7,80 +7,98 @@
...
-	 * 
+	 *
 	 * @return array $links
 	 */
 	public static function add_plugin_links( $links, $file ) {
 
-		// Check Dark Mode is the next plugin
-		if ( 'dark-mode/dark-mode.php' == $file ) {
+		// Check Dark Mode is the next plugin.
+		if ( 'dark-mode/dark-mode.php' === $file ) {
```
Comment (danieltj27) [inline dark-mode.php]: We don't need to do a comparison using `===` as `$file` is always a string.

File: dark-mode.php [hunk 7]
```diff
@@ -92,203 +110,209 @@ public static function add_plugin_links( $links, $file ) {
...
 	 * $check_auto parameter to true to check if it's set to automatically
 	 * come between two time frames and we're between the time frame now.
-	 * 
+	 *
 	 * @since 1.0
 	 * @since 1.6 Major rewrite to properly address automatic Dark Mode.
-	 * 
-	 * @param string  $user_id    User ID of given person.
+	 *
+	 * @param boolean $user_id    User ID of given person.
```
Comment (danieltj27) [inline dark-mode.php]: This has been changed to `int` as the user id should always be an integer. The check further down for the user id has also been changed to check if it's zero or not instead of `false`.

File: dark-mode.php [hunk 8]
```diff
@@ -92,203 +110,209 @@ public static function add_plugin_links( $links, $file ) {
...
+		 */
+		$css_url = apply_filters( 'dark_mode_css', plugins_url( '/', __FILE__ ) . '/darkmode.css' );
+
+		// Register the dark mode stylesheet.
+		wp_register_style( 'dark_mode', $css_url, array(), self::$version );
+
+		// Enqueue the stylesheet.
+		wp_enqueue_style( 'dark_mode' );
+
+		return true;
```
Comment (danieltj27) [inline dark-mode.php]: We don't need to return anything here, once the file is enqueued, there's no further use for it.

File: dark-mode.php [hunk 9]
```diff
@@ -297,37 +321,36 @@ public static function add_profile_fields( $profileuser ) {
...
 		$dark_mode_nonce = isset( $_POST['dark_mode_nonce'] ) ? sanitize_text_field( $_POST['dark_mode_nonce'] ) : '';
 
-		// Verify the nonce is valid
+		// Verify the nonce is valid.
 		if ( wp_verify_nonce( $dark_mode_nonce, 'dark_mode_nonce' ) ) {
 
-			// Set the value of the users choices
-			$dark_mode_core = isset ( $_POST['dark_mode'] ) ? 'on' : 'off';
-			$dark_mode_auto = isset ( $_POST['dark_mode_auto'] ) ? 'on' : 'off';
+			// Set the value of the users choices.
```
Comment (danieltj27) [inline dark-mode.php]: We don't need to indent the lines here.

File: dark-mode.php (hunk 1 above)
Comment (bueltge) [reply dark-mode.php]: THe core yes, but only in old source there is without codestyle. You should use the WP Code style, easy to use via PHP Codesniffer https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards and this rule don't support shorthand php rules. However in this context it is your plugin, your code, your rules. As example in our context, agency we use a custom ruleset because we write modern code, there need a other codestyle, we start with [PSR](https://github.com/inpsyde/php-coding-standards).

File: dark-mode.php (hunk 2 above)
Comment (bueltge) [reply dark-mode.php]: No. if you don't use a hook to include them then will WP load this in a earlier context. As example if you run a multisite, the load of a class via `new` is earlier and maybe you get error. Also include a hook for load the class is more performant, no class without no include. Maybe also helpful in this context https://wordpress.stackexchange.com/questions/63141/in-what-order-does-wordpress-load-plugin-files

File: dark-mode.php (hunk 4 above)
Comment (bueltge) [reply dark-mode.php]: I think no. If you will add tests, maybe later, a empty constructor is helpful.
Also two hints
* http://ralphschindler.com/2012/03/09/php-constructor-best-practices-and-the-prototype-pattern
* http://www.phptherightway.com/pages/Design-Patterns.html

File: dark-mode.php (hunk 5 above)
Comment (bueltge) [reply dark-mode.php]: In our context, also the WP Codex is the readability much better, if you have not more as 120 characters. Also each line is an paramter of the function `load_plugin_textdomain`.

File: dark-mode.php (hunk 6 above)
Comment (bueltge) [reply dark-mode.php]: However a strict comparison is faster, the comparison result is the same, right.

File: dark-mode.php (hunk 7 above)
Comment (bueltge) [reply dark-mode.php]: You have right. The code should use always the same typing, like string or integer. The ID should always a integer, so that `false` is not the best choice. With php7 is it much more needfully.

File: dark-mode.php (hunk 8 above)
Comment (bueltge) [reply dark-mode.php]: Same topic. The function should always get back a value, same type. In this context we get a boolean in several if checks; if all is fine, wen get an `true`. Also a design pattern for good code rules.

File: dark-mode.php (hunk 9 above)
Comment (bueltge) [reply dark-mode.php]: The change here is the dot after the comment, there is necessary for the WP Codex, see above the PHP Codesniffer ruleset.

File: dark-mode.php (hunk 1 above)
Comment (danieltj27) [reply dark-mode.php]: This is something that we could look into changing. I'll push an update at some point in the future unless you can submit a PR sooner. Happy to accept this change.

File: dark-mode.php (hunk 2 above)
Comment (danieltj27) [reply dark-mode.php]: I'll have to test this in a multi site install. What error does this produce?

File: dark-mode.php (hunk 4 above)
Comment (danieltj27) [reply dark-mode.php]: Tests won't need adding until we're ready for a merge proposal into Core and it's best not to have an empty function not doing anything.

File: dark-mode.php (hunk 6 above)
Comment (danieltj27) [reply dark-mode.php]: We could instead change this to hook into the plugin specific hook that is dynamically generated instead so we can remove this if statement.

File: dark-mode.php (hunk 7 above)
Comment (danieltj27) [reply dark-mode.php]: I've updated the plugin core to reflect this change.

**Comment by danieltj27** [comment]
I've merged a couple of changes manually into the plugin Core that you've suggested and added you as a plugin contributor, thanks 👍

**Comment by danieltj27** [comment]
Closing this as some changes have already been merged in and will be doing some major reworks to the plugins core code soon. Please open a new PR with any proposed changes again if you'd like. I appreciate the contribution 👍

**Comment by bueltge** [comment]
Hi @danieltj27 really welcome.
